### ISSUE TITLE
FocusTrapZone does not trap focus when narrator is on with scan mode and arrow keys are navigating

### ISSUE THREAD

**Issue by zachhats** [OfficeDev/office-ui-fabric-react#6535]
### Please provide a reproduction of the bug in a codepen:
 
Use the demo page in edge:
https://developer.microsoft.com/en-us/fabric#/components/focustrapzone
Turn on a focus trap zone
Turn on narrator with scan mode on (caps lock + spacebar to toggle scan mode)
From inside a FocusTrapZone, use the up or down key to navigate

### Bug Report 
 
- __Package version(s)__: current
- __Browser and OS versions__: edge- latest, windows - latest
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? No
 
Requested priority: Normal
 
Products/sites affected: PQ Online
 
#### Describe the issue:
 
Repro shown above
 
#### Actual behavior:
 
Focus can leave trap zone
 
#### Expected behavior:
 
Focus cannot leave trap zone

**Comment by Vitalius1** [comment]
Very similar bug is discussed in #6422

**Comment by zachhats** [comment]
@dzearing #6884 doesn't fix the issue, can this be reopened?

**Comment by dzearing** [comment]
This isn't something we can fix. We discussed with the Narrator team, and they recommended that:

1. We make sure we have aria-modal=true in the right places
2. They are working on a fix to respect aria-modal

So there isn't much we can do here, other than wait for them to respect the standard.

**Comment by JasonGore** [comment]
While Narrator may not accurately use `aria-modal` yet, I'm reopening this via #8654 for the following reasons:
* `FocusTrapZone` is not actually setting the `aria-modal` attribute.
* #6884 and #7654 set `aria-modal` on higher level components making use of `FocusTrapZone`, but I think `aria-modal` should have actually been set on `FocusTrapZone`. I believe `aria-modal` in these other components should be removed.

**Comment by JasonGore** [comment]
Per [discussion on the PR](https://github.com/OfficeDev/office-ui-fabric-react/pull/10559#discussion_r326800704) there are many instances where we do not want `FocusTrapZone` to act as a modal and do not want it to limit scan mode navigation. `FocusTrapZone` cannot inherently know its use case, so it's up to the users and consumers of `FocusTrapZone` to use or pass in `aria-modal` attribute as needed.
